### ISSUE TITLE
Opening the emoji picker window takes ~2 seconds

### ISSUE THREAD

**Issue by balazsorban44** [tom-james-watson/Emote#54]
Hi, thank you for your work on this!

I just installed Emote on my Ubuntu 21.10 laptop (so Wayland if relevant), and it looks great!

My one concern is when opening Emote either by my shortcut or from the terminal with the  `emote` command, it takes about **2 seconds** until the window opens.

The `README.md` suggests:

> Emote runs in the background and automatically starts when you log in.

So is this expected? Is there any limitation that makes this open so slow? Can this be related to my computer? I have a fairly high-end developer-focused Dell XPS.

The only log I can provide immediately is when opening from the terminal:

```sh
(emote:12717): dbind-WARNING **: 11:04:32.457: Couldn't connect to accessibility bus: Failed to connect to socket /tmp/dbus-aQHT5tmbfL: No such file or directory

(emote:12717): Gtk-WARNING **: 11:04:32.689: Theme parsing error: gtk.css:1427:23: 'font-feature-settings' is not a valid property name

(emote:12717): Gtk-WARNING **: 11:04:32.692: Theme parsing error: gtk.css:3317:25: 'font-feature-settings' is not a valid property name

(emote:12717): Gtk-WARNING **: 11:04:32.693: Theme parsing error: gtk.css:3779:23: 'font-feature-settings' is not a valid property name
```

I am happy to provide more information if needed. 🙂

This might be low-priority, and if there is nothing to do/there is an upstream blockage, feel free to close this issue.

**Comment by tom-james-watson** [comment]
Yeah this is far slower on Wayland, unfortunately. On X11, we register a listener in the app for a global shortcut which we can react to very quickly. On Wayland, you are basically starting a fresh instance of the program which unfortunately introduces a noticeable delay. Slow things in the chain:

- Starting a snap program
- Python
- My code somewhere?

I don't have any good ideas for how best to fix this. If I were to start this project again I would probably use have used Flatpak and Rust GTK or maybe Tauri.

The wayland hotkey issue along with being unable to auto-paste the emoji into the previous app on selection makes the app substantially worse on wayland, which is a shame. It would be nice if people could share any ideas for how to fix or work around those issues.

**Comment by balazsorban44** [comment]
I appreciate your answer and I understand. I, unfortunately, am only a user here and have little to add, but hopefully, someone coming after me will be able to give some advice.

**Comment by balazsorban44** [comment]
Getting back to this, I was wondering if it was possible to instead of closing the program on selection or Esc, we could send it to the system tray instead? Wouldn't that keep it alive then?

**Comment by cmalard** [comment]
I'm no more helpfull here, but [Guake](https://github.com/Guake/guake/) manages that by providing another command to run: `guake-toggle`
Maybe it could be a workaround? :-)
-> https://github.com/Guake/guake/blob/master/docs/source/user/faq.rst

**Comment by birgersp** [comment]
I cloned and built Emote, and then for some reason the hotkey to open it seem to actually open my locally built version. Perhaps building and running Emote has (partially) overwritten the snap install or something.

Doing the hotkey (for me CTRL+ALT+E) opens emote instantly for me now.

Opening Emote from the Linux Mint menu is still slow but I don't care. The hotkey's what's important for me 😍

**Comment by Noxmiles** [comment]
> > Getting back to this, I was wondering if it was possible to instead of closing the program on selection or `Esc`, we could send it to the system tray instead? Wouldn't that keep it alive then?
> 
> It should be possible to keep a process running with or without a dock icon / systray-like thing. And I believe this is the only real way to go around the slow startup time.

Yes, I had an other emoji picker - which is not supported anymore - which was starting with the boot and opened a window instantly. This should be also possible for emote.

But it would require the program being running in the background, only showing the GUI when the shortkeys are pressed.

**Comment by stephan-dev** [comment]
@tom-james-watson 
> register a listener in the app for a global shortcut which we can react to very quickly. 

Even on X11, this seems to be broken when we change (from Emote GUI) the main shortcut to launch emote. This 2 to 4 seconds delay appears in X11, and worse, one can get completely locked out. https://github.com/tom-james-watson/Emote/issues/87

**Comment by banan314** [comment]
I'm thinking it would be faster if I knew the way to create a `deb` or install if from sources, but it doesn't compile and I don't know how to package on Debian so I can do neither

**Comment by banan314** [comment]
and it's slow on both snap and flatpak on X11 in case anyone wondering
